test/suite/test_cursor_bound_fuzz.py

```diff
@@ -518,62 +540,45 @@ def test_bound_fuzz(self):
             prepare = random.uniform(0, 1) <= self.prepare_frequency and self.transactions_enabled
             if (self.transactions_enabled):
                 write_session.begin_transaction()
             self.apply_ops(write_session, write_cursor, prepare)
-            if (self.transactions_enabled and prepare):
-                self.verbose(2, "Preparing applied operations.")
-                write_session.prepare_transaction('prepare_timestamp=' + self.timestamp_str(self.current_ts))
-            elif (self.transactions_enabled):
-                write_session.commit_transaction('commit_timestamp=' + self.timestamp_str(self.current_ts))
+            if (self.transactions_enabled):
+                if (prepare):
+                    self.verbose(2, "Preparing applied operations.")
+                    write_session.prepare_transaction('prepare_timestamp=' + self.timestamp_str(self.current_ts))
+                else:
+                    write_session.commit_transaction('commit_timestamp=' + self.timestamp_str(self.current_ts))
+
+            # Use the current timestamp so we don't need to track previous versions.
+            if (self.transactions_enabled):
+                self.session.begin_transaction('read_timestamp=' + self.timestamp_str(self.current_ts))
+            self.run_bound_scenarios(bound_set, read_cursor)
+            if (self.transactions_enabled):
+                self.session.rollback_transaction()
+                if (prepare):
+                    write_session.commit_transaction(
+                        'commit_timestamp=' + self.timestamp_str(self.current_ts) +
+                        ',durable_timestamp='+ self.timestamp_str(self.current_ts))
             self.current_ts += 1
             if (i % 10 == 0):
                 # Technically this is a write but easier to do it with this session.
                 self.session.checkpoint()
 
-    # A lot of time was spent generating values, to achieve some amount of randomness we pre
-    # generate N values and keep them in memory.
-    value_array = []
-    iteration_count = 200 if wttest.islongtest() else 50
-    value_size = 100000 if wttest.islongtest() else 100
-    value_array_size = 20
-    key_count = 10000 if wttest.islongtest() else 1000
-    min_key = 1
-    # Max_key is not inclusive so the actual max_key is max_key - 1.
-    max_key = min_key + key_count
-    current_ts = 1
-    applied_ops = False
-    # For each iteration we do search_count searches that way we test more cases without having to
-    # generate as many key ranges.
-    search_count = 20
-    key_range = {}
-    prepare_frequency = 5/100
-    # Large transactions throw rollback errors so we don't use them in the long test.
-    transactions_enabled = False if wttest.islongtest() else True
-    update_frequency = 2/10
+        # It is possible that we are left we a prepared transaction whern we are finished. Rollback the 
```

Comment:
Also, with this refactored code flow, it looks as though the if clause on line 569 ensures that if we have a prepared transaction, it will get committed when we finish so we should never be left with a prepare transaction?